src/PrivateDns/PrivateDns/VirtualNetworkLinks/SetAzurePrivateDnsVirtualNetworkLink.cs

```diff
@@ -0,0 +1,128 @@
+        [ResourceIdCompleter("Microsoft.Network/privateDnsZones/virtualNetworkLinks")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Switch parameter that represents if registration is enabled on the link.")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter EnableRegistration { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "A hash table which represents resource tags.")]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Do not use the ETag field of the RecordSet parameter for optimistic concurrency checks.", ParameterSetName = ObjectParameterSetName)]
+        public SwitchParameter Overwrite { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+
+            PSPrivateDnsLink result = null;
+            PSPrivateDnsLink linkToUpdate = null;
+
+            switch (this.ParameterSetName)
+            {
+                case FieldsParameterSetName:
+                case ResourceParameterSetName:
+                {
+                    if (!string.IsNullOrEmpty(this.ResourceId))
+                    {
+                        PrivateDnsUtils.GetResourceGroupNameZoneNameAndLinkNameFromLinkId(this.ResourceId, out var resourceGroupName, out var zoneName, out var linkName);
+                        this.ResourceGroupName = resourceGroupName;
+                        this.ZoneName = zoneName;
+                        this.Name = linkName;
+                    }
+
+                    if (this.ZoneName.EndsWith("."))
+                    {
+                        this.ZoneName = this.ZoneName.TrimEnd('.');
+                        this.WriteWarning($"Modifying zone name to remove terminating '.'.  Zone name used is \"{this.ZoneName}\".");
+                    }
+
+                    linkToUpdate = this.PrivateDnsClient.GetPrivateDnsLink(this.Name, this.ResourceGroupName, this.ZoneName);
+                    linkToUpdate.Etag = "*";
+                    linkToUpdate.ZoneName = this.ZoneName;
+                    break;
+                }
+
+                case ObjectParameterSetName when (string.IsNullOrWhiteSpace(this.Link.Etag) || this.Link.Etag == "*") && !this.Overwrite.IsPresent:
+                    throw new PSArgumentException(string.Format(ProjectResources.Error_EtagNotSpecified, typeof(PSPrivateDnsLink).Name));
+
+                case ObjectParameterSetName:
+                    linkToUpdate = this.Link;
+                    break;
+            }
+
+            if (this.Tag != null)
+            {
+                linkToUpdate.Tags = this.Tag;
+            }
+
+            if (this.EnableRegistration.IsPresent)
+            {
+                linkToUpdate.RegistrationEnabled = true;
```

Comment:
This should be set to false if EnableRegistration is not present - that is how "Set" is expected to function.

Reply:
changed this to boolean as u mentioned on another comment.